components/org.wso2.carbon.identity.tag.manager.core/src/main/java/org/wso2/carbon/identity/tag/manager/core/dao/TagManagementDAOImpl.java

```diff
@@ -0,0 +1,361 @@
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_DELETING_ASSOCIATION.getCode(),
+                    ErrorMessage.ERROR_CODE_ERROR_DELETING_ASSOCIATION.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean isExistingAssociation(String tagUuid, String resourceId) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        boolean status;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_TAG_RESOURCE_ASC_BY_RESOURCE_TAG_UUID_SQL);
+            prepStmt.setString(1, tagUuid);
+            prepStmt.setString(1, resourceId);
+            status = prepStmt.execute();
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_GETTING_ASSOCIATION.getCode(), e);
+        }
+        return status;
+    }
+
+    @Override
+    public boolean isExistingTagId(String tagUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        boolean status;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_IDN_TAG_BY_TAG_UUID_SQL);
+            prepStmt.setString(1, tagUuid);
+            status = prepStmt.execute();
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_LOADING_TAG.getCode(), e);
+        }
+        return status;
+    }
+
+    @Override
+    public void deleteTagsByTenant(String tenantUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.DELETE_IDN_TAGS_BY_TENANT_SQL);
+            prepStmt.setString(1, tenantUuid);
+            prepStmt.execute();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_DELETING_TAGS.getCode(),
+                    ErrorMessage.ERROR_CODE_ERROR_DELETING_TAGS.getMessage(), e);
+        }
+    }
+
+    private String resolveSQLFilter(String filter) {
+
+        // To avoid any issues when the filter string is blank or null, assigning "%" to SQLFilter.
+        String sqlFilter = "%";
+        if (StringUtils.isNotBlank(filter)) {
+            sqlFilter = filter.trim()
+                    .replace("*", "%");
+        }
+        return sqlFilter;
+    }
+
+    /**
+     * Convert String value of the db stored is_publicly_visible attribute into boolean.
+     *
+     * @param isPublic Public visibility of the tag.
+     * @return The boolean value of the input String.
+     */
+    private boolean isPublic(String isPublic) {
+
+        return isPublic.equals("1");
```

Comment:
why we are having string for isPublic? Can't we use true/false for isPublic available? It is easy to understand also

Reply:
usually, we store varchar(1) for boolean. We have followed this implementing asgardeo user store